ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -112,51 +112,46 @@
   private final QueryIdGenerator queryIdGenerator;
   private final KafkaClientSupplier clientSupplier;
 
+  private final KsqlConfig ksqlConfig;
+
   public KsqlEngine(final KsqlConfig ksqlConfig) {
     this(
         new KafkaTopicClientImpl(ksqlConfig.getKsqlAdminClientConfigProps()),
         new KsqlSchemaRegistryClientFactory(ksqlConfig).create(),
         new DefaultKafkaClientSupplier(),
-        new MetaStoreImpl(new InternalFunctionRegistry())
-    );
-  }
-
-  public KsqlEngine(final KafkaTopicClient kafkaTopicClient,
-                    final SchemaRegistryClient schemaRegistryClient,
-                    final KafkaClientSupplier clientSupplier
-  ) {
-    this(
-        kafkaTopicClient,
-        schemaRegistryClient,
-        clientSupplier,
-        new MetaStoreImpl(new InternalFunctionRegistry())
+        new MetaStoreImpl(new InternalFunctionRegistry()),
+        ksqlConfig
     );
   }
 
   // called externally by tests only
   public KsqlEngine(final KafkaTopicClient topicClient,
                     final SchemaRegistryClient schemaRegistryClient,
-                    final MetaStore metaStore
+                    final MetaStore metaStore,
+                    final KsqlConfig ksqlConfig
   ) {
     this(
         topicClient,
         schemaRegistryClient,
         new DefaultKafkaClientSupplier(),
-        metaStore
+        metaStore,
+        ksqlConfig
     );
   }
 
   // called externally by tests only
   KsqlEngine(final KafkaTopicClient topicClient,
              final SchemaRegistryClient schemaRegistryClient,
              final KafkaClientSupplier clientSupplier,
-             final MetaStore metaStore
+             final MetaStore metaStore,
+             final KsqlConfig ksqlConfig
   ) {
     this.metaStore = Objects.requireNonNull(metaStore, "metaStore can't be null");
     this.topicClient = Objects.requireNonNull(topicClient, "topicClient can't be null");
     this.schemaRegistryClient =
         Objects.requireNonNull(schemaRegistryClient, "schemaRegistryClient can't be null");
     this.clientSupplier = Objects.requireNonNull(clientSupplier, "clientSupplier can't be null");
+    this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig can't be null");;
```

Comment:
Let's try and avoid having a KsqlConfig member to reduce the risk of buildMultipleQueries using the member instead of the passed-in KsqlConfig.

Reply:
Good point. I renamed it to `initializationKsqlConfig` indicating that this is the one from initialization of the engine. Let me know if you have another suggestion.